src/pages/Profile/components/UserPosts/UserPostFilter/UserPostFilter.tsx

```diff
@@ -1,5 +1,34 @@
+import { useState } from "react"
+import UserCreatePostButton from "./UserCreatePostButton"
+import UserLikePostButton from "./UserLikePostButton"
+import * as S from "./UserPostFilter.Styles"
+import { USER_POST_FILTER_LIST } from "@/pages/Profile/constants/UserPost.Constants"
+import { useParams } from "react-router-dom"
+import useAuthUserStore from "@/stores/useAuthUserStore"
+
 const UserPostFilter = () => {
-  return <div>UserPostFilter</div>
+  const { id } = useParams()
+  const { user } = useAuthUserStore()
+  const isMyPage = id === user._id
+
+  const [selectedButton, setSelectedButton] = useState(
+    USER_POST_FILTER_LIST.create,
+  )
+
+  return (
+    <S.UserPostFilterLayout $isMyPage={isMyPage}>
+      <UserCreatePostButton
+        $isSelected={selectedButton === USER_POST_FILTER_LIST.create}
+        onClick={() => setSelectedButton(USER_POST_FILTER_LIST.create)}
+      />
+      {isMyPage && (
+        <UserLikePostButton
+          $isSelected={selectedButton === USER_POST_FILTER_LIST.like}
+          onClick={() => setSelectedButton(USER_POST_FILTER_LIST.like)}
+        />
+      )}
```

Comment:
위에서 바꾸셨다면 밑에도 이렇게 바꾸시면 될 것 같습니다! 👍
```suggestion
      < UserFilteredPostButton
        $isSelected={selectedButton === USER_POST_FILTER_LIST.create}
        onClick={() => setSelectedButton(USER_POST_FILTER_LIST.create)}
        title="생성한 게시글"
      />
      {isMyPage && (
        < UserFilteredPostButton
          $isSelected={selectedButton === USER_POST_FILTER_LIST.like}
          onClick={() => setSelectedButton(USER_POST_FILTER_LIST.like)}
          title="좋아요한 게시글"
        />
      )}
```